libnetdata/socket/socket.c

```diff
@@ -1010,6 +1010,44 @@ int accept_socket(int fd, int flags, char *client_ip, size_t ipsize, char *clien
             strncpyz(client_ip, "UNKNOWN", ipsize - 1);
             strncpyz(client_port, "UNKNOWN", portsize - 1);
         }
+        char client_host[NI_MAXHOST+1];
+        if(getnameinfo((struct sockaddr *)&sadr, addrlen, client_host, (socklen_t)ipsize, NULL, 0, NI_NAMEREQD | NI_NUMERICSERV) != 0)
+            client_host[0] = 0;
+        else {
+            struct addrinfo *addr_infos = NULL;
+            if(getaddrinfo(client_host, NULL, NULL, &addr_infos)!=0) {
+                error("LISTENER: cannot validate hostname '%s' by resolving it",client_host);
+                client_host[0] = 0;
+            }
+            else {
+                struct addrinfo *scan = addr_infos;
+                int   validated = 0;
+                while(scan) {
+                    char address[INET6_ADDRSTRLEN+1];
```

Comment:
I don't think you need the `+1` here.

Reply:
Correct, it includes the null-terminator. I'll remove the +1s.